### ISSUE TITLE
Mapillary traffic sign & map feature filter

### ISSUE THREAD

**Issue by sun-geo** [openstreetmap/iD#8205]
Thanks a lot for adding the filter (date) option for photo overlays (#8133). It works very well so far. One thing i noticed, e.g. looking only to the mapillary overlay. It's seems the filter for the malliary pictures works okay, lets say only show picture-routes starting from 2019, but then when i also check the sub boxes "traffic sign" & "map feature" theses additional item on the map seems not filtered to 2019. So when i click to one of the traffic signs the small preview window opens and with a date of a picture-route e.g. from 2017. Could that please double checked and be also filter to 2019 then?

Example area: https://www.openstreetmap.org/edit#map=19/40.63242/-74.21114

**Comment by quincylvania** [comment]
The feature/sign detections should be filtered, but the issue is that the detection can be present in photos outside the set date range as well as in those within it. When you click a detection, iD arbitrarily selects one of its photos, but it should really be applying the filters first.

**Comment by nickplesha** [comment]
Just to clarify: the traffic signs and map features on the map are not filtered by the new user and date filtering. This was mainly to keep the initial filtering feature simple enough to implement but also due to the fact that the traffic signs and map features are generated from several images, often my multiple users on various dates. So the filtering there might need to work a bit differently.

I think I just need to gather more data on how users would expect these filters to apply: for example, if you filter for one particular user, should the traffic signs be shown that are from at least one image by that user or from images _exclusively_ by that user? And if you filter by date, should the map feature be shown if it was detected at least once in the date interval or if it was not detected at all outside of the date interval?

If you have any suggestions for how you'd expect these filters to work, please let me know.

**Comment by quincylvania** [comment]
> Just to clarify: the traffic signs and map features on the map are not filtered by the new user and date filtering.

@nickplesha I actually added this after your initial PR, see #8133.

This issue is about making sure iD selects a photo in the correct range when you click on a detection.

**Comment by nickplesha** [comment]
Awesome! I completely missed that. OK, I understand the issue now - when clicking on a traffic sign or map features we should make sure that the image that gets shown is one that also conforms to the active filters. To achieve this we should fetch more data about the images that are related to the traffic sign or map feature and then choose the correct one. I'll look into how to do this most efficiently with our API.

**Comment by mbrzakovic** [comment]
@nickplesha does this issue still exist after merging Mapillary v4: https://github.com/openstreetmap/iD/pull/8372?

Removing the milestone in order to speed up the next release

**Comment by nickplesha** [comment]
Yes, it still exists. I'll check if we can fix this using the new API.
